certora/specs/Protected.spec

```
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+// Check that all methods except those noted below comply with the `protected` modifier when an initiator has been set.
+rule protectedWithSetInitiator(method f, env e, calldataarg data) filtered {
+    // Do not check view functions.
+    f -> !f.isView &&
+    // Do not check the fallback function.
+    !f.isFallback &&
+    // Do not check multicall, which is used to start a new bundle.
+    f.selector != sig:multicall(bytes[]).selector
+}
+{
+    // Safe require because `protected` functions should be callable by the initiator.
+    require e.msg.sender != currentContract._initiator;
+    // Safe require because `protected` functions should be callable by Morpho.
+    require e.msg.sender != currentContract.MORPHO;
+    f@withrevert(e,data);
+    assert lastReverted;
```

this version feels more natural to me, but no strong feeling (i prefer to see the equivalence, plus you don't need to justify that it's safe requires)
```suggestion
    address initialInitiator = currentContract._initiator;
    
    f@withrevert(e,data);
    
    assert e.msg.sender != currentContract.MORPHO || require e.msg.sender != initialInitator => lastReverted;
```

---

Btw, this one is not bad neither because you write the happy path (like in the code)
```suggestion
    address initialInitiator = currentContract._initiator;
    
    f@withrevert(e,data);
    
    assert lastReverted 
        || (e.msg.sender == currentContract.MORPHO && require e.msg.sender == initialInitator);
```

---

As-is, the run [fails](https://prover.certora.com/output/976153/2d74686097c44ee3ae05efaa8fbf2abf?anonymousKey=9007c777800fb464c72a25a9f321fde25aece874) on functions that reach loops. It would require additional configuration, see [this message](https://github.com/morpho-org/morpho-blue-bundlers/pull/461#discussion_r1822778183)